metadata-ingestion/src/datahub/ingestion/source/cassandra/cassandra_profiling.py

```diff
@@ -0,0 +1,263 @@
+                entityUrn=dataset_urn, aspect=profile_aspect
+            )
+            yield mcp.as_workunit()
+
+    def populate_profile_aspect(self, profile_data: Dict) -> DatasetProfileClass:
+        field_profiles = [
+            self._create_field_profile(field_name, field_stats)
+            for field_name, field_stats in profile_data.get("column_stats", {}).items()
+        ]
+        return DatasetProfileClass(
+            timestampMillis=round(time.time() * 1000),
+            rowCount=profile_data.get("row_count"),
+            columnCount=profile_data.get("column_count"),
+            fieldProfiles=field_profiles,
+        )
+
+    def _create_field_profile(
+        self, field_name: str, field_stats: Dict
+    ) -> DatasetFieldProfileClass:
+        quantiles = field_stats.get("quantiles")
+        return DatasetFieldProfileClass(
+            fieldPath=field_name,
+            uniqueCount=field_stats.get("distinct_count"),
+            nullCount=field_stats.get("null_count"),
+            min=str(field_stats.get("min")) if field_stats.get("min") else None,
+            max=str(field_stats.get("max")) if field_stats.get("max") else None,
+            mean=str(field_stats.get("mean")) if field_stats.get("mean") else None,
+            median=str(field_stats.get("median"))
+            if field_stats.get("median")
+            else None,
+            stdev=str(field_stats.get("stdev")) if field_stats.get("stdev") else None,
+            quantiles=[
+                QuantileClass(quantile=str(0.25), value=str(quantiles[0])),
+                QuantileClass(quantile=str(0.75), value=str(quantiles[1])),
+            ]
+            if quantiles
+            else None,
+            sampleValues=field_stats.get("sample_values"),
+        )
+
+    def profile_table(
+        self, keyspace_name: str, table_name: str, columns: List[Tuple[str, str]]
+    ) -> Dict:
+
+        results: Dict[str, Any] = {}
+
+        limit = None
+        if self.config.profiling.limit:
+            limit = self.config.profiling.limit
+
+        if self.config.profiling.row_count:
+            resp = self.api.execute(
+                CassandraQueries.ROW_COUNT.format(keyspace_name, table_name), limit
+            )
+            if resp:
+                results["row_count"] = resp[0].row_count
+
+        if self.config.profiling.column_count:
+            resp = self.api.execute(
+                CassandraQueries.COLUMN_COUNT.format(keyspace_name, table_name), limit
+            )
+            if resp:
+                results["column_count"] = resp[0].column_count
+
+        if not self.config.profiling.profile_table_level_only:
+            resp = self.api.execute(
+                f'SELECT {", ".join([col[0] for col in columns])} FROM {keyspace_name}."{table_name}"',
+                limit,
+            )
+            results["column_metrics"] = resp
+
+        return self._parse_profile_results(results, columns)
```

Comment:
If I understand correctly, we fetch `limit` number of rows and compute profiling metrics in python code here ?

Reply:
Yes